mobile/lib/modules/search/providers/people.provider.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/modules/home/ui/asset_grid/asset_grid_data_structure.dart';
+import 'package:immich_mobile/modules/search/services/search.service.dart';
+import 'package:immich_mobile/shared/models/asset.dart';
+import 'package:openapi/api.dart';
+
+final personAssetsProvider = FutureProvider.family
+    .autoDispose<RenderList, String>((ref, personId) async {
+  final SearchService searchService = ref.watch(searchServiceProvider);
+
+  var assets = await searchService.getPersonAssets(personId);
+
+  if (assets == null) {
+    return RenderList.empty();
+  }
+
+  var convertToAsset = assets.map((e) => Asset.remote(e)).toList();
+
+  return RenderList.fromAssets(convertToAsset, GroupAssetsBy.auto);
+});
```

Comment:
@fyfrey What do you think about this approach, I was going to go with syncing the `people` property on each asset and use it to query from Isar, but I am not sure how to tackle the situation where the property is an array

Here is the example AssetDtoReponse with the new `people` property

```JSON
{
"id": "450352af-d4bc-42a7-8123-823382241ca5",
"deviceAssetId": "web-1_9eB_32e8YEq0jrkKpRs2ng.0.jpeg-1684270708686",
"ownerId": "52961026-9726-43d5-a10b-3f2c5bf9a7f9",
"deviceId": "WEB",
"type": "IMAGE",
"originalPath": "upload/library/admin/2023/05/16/2023-05-16-1_9eB_32e8YEq0jrkKpRs2ng.0.jpeg",
"originalFileName": "1_9eB_32e8YEq0jrkKpRs2ng.0",
"resizePath": "upload/thumbs/52961026-9726-43d5-a10b-3f2c5bf9a7f9/450352af-d4bc-42a7-8123-823382241ca5.jpeg",
"fileCreatedAt": "2023-05-16T20:58:28.686Z",
"fileModifiedAt": "2023-05-16T20:58:28.686Z",
"updatedAt": "2023-05-23T09:49:11.290Z",
"isFavorite": false,
"isArchived": false,
"mimeType": "image/jpeg",
"webpPath": "upload/thumbs/52961026-9726-43d5-a10b-3f2c5bf9a7f9/450352af-d4bc-42a7-8123-823382241ca5.webp",
"encodedVideoPath": null,
"duration": "0:00:00.000000",
"exifInfo": {
"make": null,
"model": null,
"exifImageWidth": 1400,
"exifImageHeight": 1400,
"fileSizeInByte": 847421,
"orientation": null,
"dateTimeOriginal": "2023-05-16T20:58:28.686Z",
"modifyDate": "2023-05-16T20:58:28.686Z",
"timeZone": null,
"lensModel": null,
"fNumber": null,
"focalLength": null,
"iso": null,
"exposureTime": null,
"latitude": null,
"longitude": null,
"city": null,
"state": null,
"country": null,
"description": ""
},
"livePhotoVideoId": null,
"people": [
{
"id": "4637871f-3840-4f02-9e5c-be0275a5894f",
"name": "Famous Astrologist",
"thumbnailPath": "upload/thumbs/52961026-9726-43d5-a10b-3f2c5bf9a7f9/4637871f-3840-4f02-9e5c-be0275a5894f.jpeg"
}
]
},
```

Reply:
Ah we can use `Backlink` similar to the relationship between `User` and `Album`, right?

Reply:
yeah, we have three options:

1. use IsarLinks. we currently don't have any links on `Asset` because of performance reasons and we cannot pass `Asset` to other dart isolates. we no longer do this since my lazy loading PR. so we actually could use IsarLinks now
2. Use embedded objects https://isar.dev/schema.html#embedded-objects to store all person information as in array inline (probably also bad for performance)
3. Use an array of people IDs similar to how `asset.dart` has `ownerId` referring to a `User`.